src/lib/ecs/system.ts

```diff
@@ -348,7 +359,21 @@ export class SystemManager {
 		}
 
 		if (!success) {
-			task.spawn(error, `System: ${system.name} errored! ${result} + \n ${debug.traceback}`);
+			if (system.recentErrors === undefined) {
+				system.recentErrors = new Map();
+			}
+
+			const recentError = `System: ${system.name} errored! ${result} + \n ${debug.traceback}`;
+
+			const lastError = system.recentErrors.get(recentError);
+
+			if (lastError !== undefined && lastError > os.clock()) {
+				return;
+			}
+
+			system.recentErrors.set(recentError, os.clock() + 10);
+			task.spawn(error, recentError);
+			warn("The above error has been suppressed for 10 seconds.");
```

Comment:
Can we do this some other way? Seems like useful information to know when/how much things are erroring; this is the original idea between flares but I never implemented the necessary UI for it.

Reply:
Because nearly every system will run every frame, most of these errors will likely occur every frame everytime the system is executed. FYI this is based off matters implementation. I am happy to do it another way if you can think of one, but this was the reason I did it this way for now - I could have completely just unscheduled the system if there was an error but this would be harder to fix the issues if they arose during hotreloading.